odoo/addons/base/models/ir_ui_view.py

```diff
@@ -108,21 +108,33 @@ def transfer_node_to_modifiers(node, modifiers, context=None):
         else:
             modifiers['invisible'] = [('state', 'not in', states)]
 
+    postprocess_context_modifiers = []
     for attr in ('invisible', 'readonly', 'required'):
         value_str = node.attrib.pop(attr, None)
         if value_str:
-            value = bool(quick_eval(value_str, {'context': context or {}}))
+
             if (attr == 'invisible'
                     and any(parent.tag == 'tree' for parent in node.iterancestors())
                     and not any(parent.tag == 'header' for parent in node.iterancestors())):
                 # Invisible in a tree view has a specific meaning, make it a
                 # new key in the modifiers attribute.
-                modifiers['column_invisible'] = value
-            elif value or (attr not in modifiers or not isinstance(modifiers[attr], list)):
+                attr = 'column_invisible'
+
+            # TODO: for invisible="context.get('...')", delegate to the web client.
+            value = quick_eval(value_str, {'context': context}, evaluate=False)
+            if isinstance(value, str):
+                postprocess_context_modifiers.append(attr)
+            else:
+                value = bool(value)
+
+            if value or (attr not in modifiers or not isinstance(modifiers[attr], list)):
                 # Don't set the attribute to False if a dynamic value was
                 # provided (i.e. a domain from attrs or states).
                 modifiers[attr] = value
 
+    if postprocess_context_modifiers:
+        node.set('postprocess_context_modifiers', ','.join(postprocess_context_modifiers))
```

Comment:
Name's not great, maybe `delayed-modifiers`, `unevaluated-modifiers` or `context-dependent-modifiers`? So their purpose is clear.

Reply:
Agreed as well, switched to `context-dependent-modifiers`